AlphaWallet/Browser/Coordinators/ScanQRCodeCoordinator.swift

```diff
@@ -51,26 +120,200 @@ final class ScanQRCodeCoordinator: NSObject, Coordinator {
     }
 
     @objc func dismiss() {
-        navigationController.dismiss(animated: true) {
+        stopScannerAndDissmiss {
             self.delegate?.didCancel(in: self)
         }
     }
+
+    func resolveScanResult(_ rawValue: String) {
+        guard let delegate = resolutionDelegate else { return }
+
+        switch ScanQRCodeResolution(rawValue: rawValue) {
+        case .value(let value):
+            switch value {
+            case .address(let contract):
+                switch configuration {
+                case .resolution(let tokensDatastores, _):
+                    let actions: [ScanQRCodeAction]
+                    //NOTE: or meybe we need pass though all servers?
+                    guard let tokensDatastore = tokensDatastores.first(where: { $0.server == rpcServer }) else { return }
+
+                    //I guess if we have token, we shouldn't be able to send to it, or we should?
+                    if tokensDatastore.token(forContract: contract) != nil {
```

Comment:
We should still allow `.sendToAddress` and `.watchWallet` because for the former, some token-holding contracts (I think Dai?) works by getting you to send ETH to it. For the latter, we might just want to look at transactions for the contract.